File: Python/aboveaverage.py
'''
 *
 *	Problem: "Above Average"
 *
'''

numCases = int(input())

for case in range(numCases):

	grades = input().split(" ")
	numGrades = int(grades[0])
	
	sumGrades = 0
	for index in range(numGrades):
		gradeIndex = index + 1
		grade = int(grades[gradeIndex])
		sumGrades += grade
		
	average = sumGrades / numGrades
	
	aboveAverage = 0
	
	for index in range(numGrades):
		gradeIndex = index + 1
		grade = int(grades[gradeIndex])
		if (grade > average):
			aboveAverage += 1		
	
	percAboveAverage = float(100 * (aboveAverage / numGrades))
	
	formattedResult = float("{:.3f}".format(percAboveAverage))
	print("%.3f%%" % (percAboveAverage))

File: Ruby/aboveaverage.rb
=begin
 *
 *	Problem:		Above Average
 *	Problem ID:	aboveaverage
 *	url:				https://open.kattis.com/problems/aboveaverage
 *	Solution:		Ruby
 *
=end

# above_average
def above_average(grades, average)
	above_average = 0
	grades.each do |grade|
		if (grade > average)
			above_average += 1
		end
	end
	ratio =  (above_average.to_f / grades.size) * 100
	return ratio
end # end above_average

# main
num_cases = gets.chomp.to_i
for current_case in 1..num_cases
	case_data = gets.chomp.split(" ")
	num_students = case_data[0].to_i
	grades = case_data.slice( 1, (case_data.size - 1) ).map(&:to_i)
	
	total_grades = grades.sum	
	average = total_grades.to_f / num_students
	
	result = above_average(grades, average)
	puts "%.3f%%" % result.round(3)
end